substrate/frame/message-queue/src/lib.rs

```diff
@@ -904,6 +919,20 @@ impl<T: Config> Pallet<T> {
 					Self::single_msg_overhead(),
 				))
 			}
+
+			if service.any_gt(max_block) {
+				return Err(format!(
+					"ServiceWeight {service} is bigger than max block weight {max_block}"
+				))
+			}
+		}
+
+		if let Some(on_idle) = T::IdleMaxServiceWeight::get() {
+			if on_idle.any_gt(max_block) {
```

Comment:
There could also be a check here that either `T::ServiceWeight::get().all_gt(T::IdleMaxServiceWeight::get()) || T::IdleMaxServiceWeight::get().all_gt(T::ServiceWeight::get())` to ensure that the `any_gt` in the `max_message_weight` function will for sure find the larger one.  

Otherwise one could be larger in ref time and the other larger in proof.

Reply:
Yeah I also thought about this. I hope people don't do this, but I can add another check.